transport.go

```diff
@@ -245,3 +344,7 @@ func (t *transport) Protocols() []int {
 func (t *transport) String() string {
 	return "QUIC"
 }
+
+func holePunchKey(addr net.Addr, id peer.ID) string {
```

Comment:
nit: we can also just use a custom type `holePunchKey{addr string, id peer.ID}`. Not critical, but that would be slightly easier to reason about.